beacon-chain/blockchain/core.go

```diff
@@ -0,0 +1,71 @@
+package blockchain
+
+import (
+	"sync"
+
+	"github.com/ethereum/go-ethereum/ethdb"
+	"github.com/ethereum/go-ethereum/rlp"
+	"github.com/prysmaticlabs/geth-sharding/beacon-chain/types"
+	log "github.com/sirupsen/logrus"
+)
+
+var stateLookupKey = "beacon-chain-state"
+
+// BeaconChain represents the core PoS blockchain object containing
+// both a crystallized and active state.
+type BeaconChain struct {
```

Comment:
what about beacon chain block? should it be in `BeaconChain`?

Reply:
BeaconChain will most likely follow the structure of geth where we have Blockchain object that stores items such as the latest block header and has methods to retrieve blocks and grow the chain.